components/metadata-service/internal/httperrors/httperrors.go

```diff
@@ -28,7 +33,17 @@ func errorCodeToHttpStatus(code int) int {
 	}
 }
 
-func AppErrorToResponse(appError apperrors.AppError) (status int, body ErrorResponse) {
-	httpCode := errorCodeToHttpStatus(appError.Code())
-	return httpCode, ErrorResponse{httpCode, appError.Error()}
+func formatErrorResponse(httpCode int, errorMessage string, detailedErrorResponse bool) (status int, body ErrorResponse) {
+	if isInternalError(httpCode) && !detailedErrorResponse {
+		return httpCode, ErrorResponse{httpCode, "Internal error."}
+	}
+	return httpCode, ErrorResponse{httpCode, errorMessage}
+}
+
+func isInternalError(httpCode int) bool {
```

Comment:
You could simplify this function:
func isInternalError(httpCode int) bool {
   return httpCode == http.StatusInternalServerError
}

Reply:
Done!